TestTopFieldCollector.testTotalHits test failure

### Description

TestTopFieldCollector.testTotalHits fails on branch_9x, git-bisect indicates https://github.com/apache/lucene/commit/0aa88910ca9a1032d288996d14203eac4953f2de

I tried reproducing on `main` but it didn't fail

```
org.apache.lucene.search.TestTopFieldCollector > testTotalHits FAILED
    java.lang.AssertionError: expected:<2> but was:<3>
        at __randomizedtesting.SeedInfo.seed([71EA3D3296E5473E:5B9A6F84977235B9]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.apache.lucene.search.TestTopFieldCollector.testTotalHits(TestTopFieldCollector.java:194)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
```

### Gradle command to reproduce

./gradlew test --tests TestTopFieldCollector.testTotalHits -Dtests.seed=71EA3D3296E5473E -Dtests.locale=en-HK -Dtests.timezone=Australia/West -Dtests.asserts=true -Dtests.file.encoding=UTF-8

--- a/lucene/core/src/test/org/apache/lucene/search/TestTopFieldCollector.java
+++ b/lucene/core/src/test/org/apache/lucene/search/TestTopFieldCollector.java
@@ -31,6 +31,7 @@ import org.apache.lucene.document.TextField;
 import org.apache.lucene.index.DirectoryReader;
 import org.apache.lucene.index.IndexReader;
 import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
 import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.index.NoMergePolicy;
 import org.apache.lucene.index.Term;
@@ -180,7 +181,12 @@ public class TestTopFieldCollector extends LuceneTestCase {
     Sort sort = new Sort(new SortField("foo", SortField.Type.LONG));
     IndexWriter w =
         new IndexWriter(
-            dir, newIndexWriterConfig().setMergePolicy(NoMergePolicy.INSTANCE).setIndexSort(sort));
+            dir,
+            newIndexWriterConfig()
+                .setMergePolicy(NoMergePolicy.INSTANCE)
+                .setIndexSort(sort)
+                .setMaxBufferedDocs(7)
+                .setRAMBufferSizeMB(IndexWriterConfig.DISABLE_AUTO_FLUSH));
     Document doc = new Document();
     doc.add(new NumericDocValuesField("foo", 3));
     for (Document d : Arrays.asList(doc, doc, doc, doc)) {

